src/include/daos_srv/control.h

```diff
@@ -162,4 +183,53 @@ struct nvme_stats {
  * \return		Zero on success, negative value on error
  */
 int copy_ascii(char *dst, size_t dst_sz, const void *src, size_t src_sz);
+
+/**
+ * Check acceleration engine setting is valid.
+ *
+ * \param[in]  in	input acceleration engine string buffer
+ * \param[out] result	true if input matches a known setting, false otherwise
+ *
+ * \return		non-zero on error, zero otherwise
+ */
+static inline int
+nvme_conf_validate_accel_engine(char *in, bool *result)
```

Comment:
Is this being used by C code somewhere, or just pulled into CGo? Would be simpler to write in Go.

Reply:
the thinking is that the logic should reside in C code so that someone updating optional capabilities in the engine shouldn't need to update control-plane code, that's kind of not true at the moment anyway (because the optmask requires a list of flags) but moving these functions into go would mean that when we add capabilities we have to update go code is that acceptable @jolivier23 ?

Reply:
I've moved them into Go as requested, done.